src/service/core-services.js

```diff
@@ -125,12 +140,16 @@ function installAmpdocServicesInternal(ampdoc, isEmbedded) {
     ? adoptServiceForEmbedDoc(ampdoc, 'history')
     : installHistoryServiceForDoc(ampdoc);
 
-  isEmbedded
-    ? installInaboxResourcesServiceForDoc(ampdoc)
-    : installResourcesServiceForDoc(ampdoc);
+  if (isShadow) {
+    adoptServiceForEmbedDoc(ampdoc, 'resources');
```

Comment:
This might need to use `installResourcesServiceForDoc`, we'll need verify that the next-page's resources are properly cleaned up when the next-page is removed.